source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -3950,6 +3978,10 @@ export abstract class Action extends CardObject {
         return this.internalValidateInputs(this.getReferencedInputs());
     }
 
+    get shouldPromoteAsPrimaryOnExecute(): boolean {
+        return this.mode === "primary";
+    }
+
     get isPrimary(): boolean {
```

Comment:
is this getter still accurately named? should we be checking `mode` here? (entirely possible that I'm missing something)

Reply:
(in case it's not clear, I'm referring to `isPrimary()`)

Reply:
`isPrimary` is something that's existed in the JS API for a while, and is unrelated. Outlook uses it, I can't rename it unfortunately. So it's not great, but I have no better option other than to suggest different names for the mode values. Maybe "main" and "secondary"

Reply:
perhaps just some docs on `isPrimary()` to clarify its purpose?